libvast/include/vast/detail/inspection_common.hpp

```diff
@@ -0,0 +1,96 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include <string_view>
+#include <type_traits>
+#include <utility>
+
+namespace vast::detail {
+
+struct NoOpCallback {
+  constexpr bool operator()() noexcept {
+    return true;
+  }
+};
+
+/// A Common class used by VAST inspectors as return type for the object method.
+/// The CAF inspector API requires inspectors to have an object method.
+/// This method should return an object which guides the inspection procedure.
+/// Example of usage:
+/// bool inspect(Inspector& f, some_type& x)
+///{
+///   auto cb = []{};
+///   return f.object(x).
+///     on_load(cb).
+///     fields(f.field("name", x.name), f.field("value", x.value));
+///}
+/// Such code should result in inspection of x.name by the inspector. If the
+/// inspection succeeds then it should proceed to inspect x.value. If all
+/// provided fields succeed then the callback `cb` should be called.
+template <class Inspector, class AfterInspectionCallback = NoOpCallback>
```

Comment:
```suggestion
template <class Inspector, class AfterInspectionCallback = decltype([] {
                             return true;
                           })>
```

You might argue that this is less readable and I agree, but returning `true` from a "NoOp" is pretty random and it makes it less likely to be useful anywhere else.